File: react-vite/src/mockData/foodListings.js
// Mock data for food listings
export const mockFoodListings = [
    {
        id: 1,
        provider_id: 1,
        title: "Fresh Organic Vegetables",
        description: "Assorted organic vegetables including carrots, broccoli, and lettuce",
        food_type: "PRODUCE",
        quantity: 25.5,
        unit: "kg",
        expiration_date: "2024-03-25T15:00:00Z",
        status: "available",
        allergens: [],
        created_at: "2024-03-20T10:00:00Z",
        image_url: "/vegetables.jpg"
    },
    {
        id: 2,
        provider_id: 1,
        title: "Artisan Bread Selection",
        description: "Freshly baked artisan breads including sourdough and whole wheat",
        food_type: "BAKERY",
        quantity: 40,
        unit: "pieces",
        expiration_date: "2024-03-22T18:00:00Z",
        status: "pending",
        allergens: ["gluten", "wheat"],
        created_at: "2024-03-20T09:30:00Z",
        image_url: "/bread.jpg"
    },
    {
        id: 3,
        provider_id: 1,
        title: "Mixed Fruit Basket",
        description: "Seasonal fruits including apples, oranges, and bananas",
        food_type: "PRODUCE",
        quantity: 15,
        unit: "kg",
        expiration_date: "2024-03-23T12:00:00Z",
        status: "available",
        allergens: [],
        created_at: "2024-03-20T11:15:00Z",
        image_url: "/fruits.jpg"
    },
    {
        id: 4,
        provider_id: 1,
        title: "Dairy Products",
        description: "Assorted dairy including milk, yogurt, and cheese",
        food_type: "DAIRY",
        quantity: 30,
        unit: "items",
        expiration_date: "2024-03-24T10:00:00Z",
        status: "completed",
        allergens: ["dairy", "lactose"],
        created_at: "2024-03-19T14:20:00Z",
        image_url: "/dairy.jpg"
    },
    {
        id: 5,
        provider_id: 1,
        title: "Canned Goods",
        description: "Various canned vegetables and fruits",
        food_type: "PANTRY",
        quantity: 50,
        unit: "cans",
        expiration_date: "2024-06-20T00:00:00Z",
        status: "available",
        allergens: [],
        created_at: "2024-03-20T08:45:00Z",
        image_url: "/canned.jpg"
    },
    {
        id: 6,
        provider_id: 1,
        title: "Fresh Pasta",
        description: "Handmade fresh pasta assortment",
        food_type: "PREPARED",
        quantity: 20,
        unit: "kg",
        expiration_date: "2024-03-22T20:00:00Z",
        status: "pending",
        allergens: ["gluten", "eggs"],
        created_at: "2024-03-20T13:00:00Z",
        image_url: "/pasta.jpg"
    },
    {
        id: 7,
        provider_id: 1,
        title: "Frozen Meals",
        description: "Pre-prepared frozen meals",
        food_type: "FROZEN",
        quantity: 35,
        unit: "meals",
        expiration_date: "2024-04-20T00:00:00Z",
        status: "available",
        allergens: ["soy", "dairy"],
        created_at: "2024-03-20T15:30:00Z",
        image_url: "/frozen.jpg"
    },
    {
        id: 8,
        provider_id: 1,
        title: "Snack Pack",
        description: "Assorted healthy snacks and nuts",
        food_type: "SNACKS",
        quantity: 100,
        unit: "packs",
        expiration_date: "2024-04-15T00:00:00Z",
        status: "available",
        allergens: ["nuts", "peanuts"],
        created_at: "2024-03-20T16:45:00Z",
        image_url: "/snacks.jpg"
    }
];

// Helper function to get listings by status
export const getListingsByStatus = (status) => {
    return mockFoodListings.filter(listing => listing.status === status);
};

// Helper function to get listings by food type
export const getListingsByType = (foodType) => {
    return mockFoodListings.filter(listing => listing.food_type === foodType);
};

// Helper function to calculate metrics
export const calculateMetrics = () => {
    return {
        total: mockFoodListings.length,
        available: getListingsByStatus('available').length,
        pending: getListingsByStatus('pending').length,
        completed: getListingsByStatus('completed').length
    };
};

// Helper function to get recent listings
export const getRecentListings = (limit = 5) => {
    return [...mockFoodListings]
        .sort((a, b) => new Date(b.created_at) - new Date(a.created_at))
        .slice(0, limit);
};

File: app/seeds/food_listings.py
from app.models import db, FoodListing, Provider, FoodType, FoodStatus, environment, SCHEMA
from sqlalchemy.sql import text
from datetime import datetime, timedelta

def seed_food_listings():
    print("\n=== Starting Food Listing Seeding ===")
    
    if environment == "production":
        db.session.execute(f"TRUNCATE table {SCHEMA}.food_listings RESTART IDENTITY CASCADE;")
    else:
        db.session.execute(text("DELETE FROM food_listings"))
    
    providers = Provider.query.all()
    print(f"Found {len(providers)} providers")
    
    listings = []
    for provider in providers:
        print(f"Creating listings for provider {provider.id}")
        listings.extend([
            FoodListing(
                provider_id=provider.id,
                title="Fresh Mixed Vegetables",
                description="Assorted fresh vegetables including carrots, broccoli, and bell peppers",
                food_type=FoodType.PRODUCE,
                quantity=5.0,
                unit="kg",
                original_price=25.99,
                discounted_price=12.99,
                expiration_date=get_dates(5)[0],
                best_by_date=get_dates(4)[0],
                pickup_window_start=get_dates(5)[1],
                pickup_window_end=get_dates(5)[2],
                allergens=["None"],
                is_perishable=True,
                storage_instructions="Keep refrigerated",
                handling_instructions="Handle with care, wash before use",
                temperature_requirements="34-40°F",
                image_url="/veggie.jpg",
                status=FoodStatus.AVAILABLE
            ),
            FoodListing(
                provider_id=provider.id,
                title="Fresh Chicken Breasts",
                description="Fresh, never frozen chicken breasts perfect for grilling",
                food_type=FoodType.MEAT,
                quantity=3.0,
                unit="kg",
                original_price=35.99,
                discounted_price=17.99,
                expiration_date=get_dates(3)[0],
                best_by_date=get_dates(2)[0],
                pickup_window_start=get_dates(3)[1],
                pickup_window_end=get_dates(3)[2],
                allergens=["None"],
                is_perishable=True,
                storage_instructions="Keep frozen or refrigerated",
                handling_instructions="Keep at safe temperature",
                temperature_requirements="Below 40°F",
                image_url="/chicken.jpg",
                status=FoodStatus.AVAILABLE
            ),
            FoodListing(
                provider_id=provider.id,
                title="Fresh Milk Gallons",
                description="Fresh whole milk, perfect for families",
                food_type=FoodType.DAIRY,
                quantity=10,
                unit="gallons",
                original_price=45.99,
                discounted_price=22.99,
                expiration_date=get_dates(7)[0],
                best_by_date=get_dates(5)[0],
                pickup_window_start=get_dates(7)[1],
                pickup_window_end=get_dates(7)[2],
                allergens=["Milk"],
                is_perishable=True,
                storage_instructions="Keep refrigerated",
                handling_instructions="Handle with care",
                temperature_requirements="Below 40°F",
                image_url="/milk.avif",
                status=FoodStatus.AVAILABLE
            ),
            FoodListing(
                provider_id=provider.id,
                title="Fresh Baked Pastries",
                description="Assorted freshly baked pastries including croissants and danishes",
                food_type=FoodType.BAKERY,
                quantity=20,
                unit="pieces",
                original_price=40.00,
                discounted_price=18.99,
                expiration_date=get_dates(2)[0],
                best_by_date=get_dates(1)[0],
                pickup_window_start=get_dates(2)[1],
                pickup_window_end=get_dates(2)[2],
                allergens=["Wheat", "Milk", "Eggs"],
                is_perishable=True,
                storage_instructions="Store in a cool, dry place",
                handling_instructions="Handle with care",
                temperature_requirements="Room temperature",
                image_url="/pastry.png",
                status=FoodStatus.AVAILABLE
            ),
            FoodListing(
                provider_id=provider.id,
                title="Canned Goods Assortment",
                description="Mixed canned vegetables, fruits, and beans",
                food_type=FoodType.PANTRY,
                quantity=15,
                unit="cans",
                original_price=30.00,
                discounted_price=14.99,
                expiration_date=get_dates(365)[0],
                best_by_date=get_dates(300)[0],
                pickup_window_start=get_dates(30)[1],
                pickup_window_end=get_dates(30)[2],
                allergens=["None"],
                is_perishable=False,
                storage_instructions="Store in a cool, dry place",
                handling_instructions="Check cans for dents or damage",
                temperature_requirements="Room temperature",
                image_url="/canned.jpg",
                status=FoodStatus.AVAILABLE
            ),
            FoodListing(
                provider_id=provider.id,
                title="Prepared Meal Boxes",
                description="Ready-to-eat meal boxes with balanced nutrition",
                food_type=FoodType.PREPARED,
                quantity=8,
                unit="boxes",
                original_price=64.00,
                discounted_price=29.99,
                expiration_date=get_dates(4)[0],
                best_by_date=get_dates(3)[0],
                pickup_window_start=get_dates(4)[1],
                pickup_window_end=get_dates(4)[2],
                allergens=["Milk", "Wheat", "Soy"],
                is_perishable=True,
                storage_instructions="Keep refrigerated",
                handling_instructions="Reheat thoroughly before consuming",
                temperature_requirements="Below 40°F",
                image_url="/meal-box.jpg",
                status=FoodStatus.AVAILABLE
            ),
            FoodListing(
                provider_id=provider.id,
                title="Assorted Snacks",
                description="Mixed healthy snacks including granola bars and dried fruits",
                food_type=FoodType.OTHER,
                quantity=25,
                unit="packages",
                original_price=50.00,
                discounted_price=24.99,
                expiration_date=get_dates(90)[0],
                best_by_date=get_dates(60)[0],
                pickup_window_start=get_dates(30)[1],
                pickup_window_end=get_dates(30)[2],
                allergens=["Nuts", "Soy"],
                is_perishable=False,
                storage_instructions="Store in a cool, dry place",
                handling_instructions="Check package integrity",
                temperature_requirements="Room temperature",
                image_url="/snck.jpg",
                status=FoodStatus.AVAILABLE
            )
        ])
    
    try:
        for listing in listings:
            db.session.add(listing)
            print(f"Adding listing: {listing.title} with status {listing.status}")
        
        db.session.commit()
        print(f"Successfully seeded {len(listings)} food listings")
        
        # Verify listings
        all_listings = FoodListing.query.all()
        for listing in all_listings:
            print(f"Verified listing {listing.id}: {listing.title} - Status: {listing.status}")
        
        return listings
    except Exception as e:
        print(f"Error seeding food listings: {str(e)}")
        db.session.rollback()
        raise e

def undo_food_listings():
    if environment == "production":
        db.session.execute(f"TRUNCATE table {SCHEMA}.food_listings RESTART IDENTITY CASCADE;")
    else:
        db.session.execute(text("DELETE FROM food_listings"))
    db.session.commit()

# Helper function to create datetime objects
def get_dates(days_until_expiry=7, pickup_window_hours=48):
    now = datetime.now()
    expiry = now + timedelta(days=days_until_expiry)
    pickup_start = now + timedelta(hours=1)
    pickup_end = pickup_start + timedelta(hours=pickup_window_hours)
    return expiry, pickup_start, pickup_end 